packages/apps/crm/src/components/templates/ValidateButtonOpportunity/ValidateButtonOpportunity.js

```diff
@@ -0,0 +1,112 @@
+/*
+ * Axelor Business Solutions
+ *
+ * Copyright (C) 2023 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or  modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, {useCallback} from 'react';
+import {StyleSheet, View} from 'react-native';
+import {useDispatch, useSelector, useTranslator} from '@axelor/aos-mobile-core';
+import {Button} from '@axelor/aos-mobile-ui';
+import {
+  createOpportunity,
+  updateOpportunity,
+} from '../../../features/opportunitySlice';
+
+const ValidateButtonOpportunity = ({
+  _opportunity,
+  company,
+  navigation,
+  opportunity,
+  idOpportunity,
+  disabled,
+}) => {
+  const I18n = useTranslator();
+  const dispatch = useDispatch();
+
+  const {user} = useSelector(state => state.user);
+
+  const createOpportinityAPI = useCallback(() => {
+    dispatch(
+      createOpportunity({
+        opportunity: {
+          ..._opportunity,
+          user: {id: user.id},
+          partner: {id: _opportunity.partner?.id},
+          contact: {id: _opportunity.contact?.id},
+          name: _opportunity.partner?.fullName,
+          amount: _opportunity.amount != null ? _opportunity.amount : 0,
+          recurrentAmount:
+            _opportunity.recurrentAmount != null
+              ? _opportunity.recurrentAmount
+              : 0,
+          probability: '0',
+          worstCase: '0',
+          expectedDurationOfRecurringRevenue: 0,
+          bestCase: '0',
```

Comment:
Those checks won't be necessary anymore once opportunity object is correctly initialized